### ISSUE TITLE
[BUG]: Cannot import Python module tokenizers 

### ISSUE THREAD

**Issue by johnhuichen** [modularml/mojo#854]
### Bug description

I was unable to import Python module tokenizers (https://huggingface.co/docs/tokenizers/index) using Mojo

### Steps to reproduce

1. Create a Python virtual environment if you'd like, I used Anaconda (conda create --name test python=3.11)
2. Install tokenizers (pip install tokenizers)
3. Create a new file main.mojo and copy paste the following code
```
from python import Python

fn main():
  try:
    let my_module = Python.import_module("tokenizers")
  except e:
    print(e.value)
```
4. Run `mojo main.mojo` in terminal and get a generic error message: "An error occurred in Python"

![Screenshot from 2023-09-19 16-32-56](https://github.com/modularml/mojo/assets/15637790/edca1a20-bf93-483f-9d9b-33fe8ecf5c9f)

![Screenshot from 2023-09-19 16-33-09](https://github.com/modularml/mojo/assets/15637790/be11cebe-62f6-4f70-86a2-3ec96788d6f4)

### System information

```shell
- What OS did you do install Mojo on ?
Ubuntu 22.04.3 LTS

- Provide version information for Mojo by pasting the output of `mojo -v`
mojo 0.2.1 (64d14e85)

- Provide Modular CLI version by pasting the output of `modular -v`
modular 0.1.4 (6b54d308)
```

**Comment by gujuriu** [comment]
As a follow up for this bug, I managed to find a work around but does not work with Python 3.11, it works with Python 3.10 or 3.9.

Initially I've installed tokenizers in the non-conda env (like... directly in the Python 3.10.2, this is either Python from my Ubuntu 22.04.2 LTS or what comes from Mojo installation, I'm not sure), I've executed your code and works just fine (I'm not saying is not a bug what you posted, maybe it's a limitation of current version of Mojo, or just a bug).

Then I've created a new conda env with a different minor version of Python (3.9.18) `conda create -n p3.9.18 python=3.9.18`, I've activated the new env `conda activate p3.9.18`, installed tokenizers `pip install tokenizers==0.12.1`, this time with a slightly older version... to make sure is not loading (for some reason) from base env, and then I've executed this line `export MOJO_PYTHON_LIBRARY=~/miniconda3/envs/p3.9.18/lib/libpython3.so` (it might not be the right way but is doing the job for me, also I'm using miniconda) followed by running your code from above (I've just added `print(my_module.__version__)` after the line with `let my_module = Python.import_module("tokenizers")` to see what's loading, saved the content in file `u.mojo` and then executed `mojo u.mojo` and then I got `0.12.1` as output.

For your main example (creating a conda env for Python 3.11, activate it, installing tokenizers in that env and then running the mojo code in it), I'm getting this, but only if I export (like I did above) the path to libpython3.so of my current env, e.g. in my case `export MOJO_PYTHON_LIBRARY=~/miniconda3/envs/p3.11.4/lib/libpython3.so`
now, if I try to run the same code from above, I'm getting this:
```
$ export MOJO_PYTHON_LIBRARY=~/miniconda3/envs/p3.11.4/lib/libpython3.so
$ mojo u.mojo
[34138:34138:20230921,134203.635412:ERROR file_io_posix.cc:144] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq: No such file or directory (2)
[34138:34138:20230921,134203.635440:ERROR file_io_posix.cc:144] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq: No such file or directory (2)
[34138:34139:20230921,134203.635946:ERROR directory_reader_posix.cc:42] opendir /home/<my_user_dir>/.modular/crashdb/attachments/17de0001-e526-451b-b51f-88ce8283cc74: No such file or directory (2)
Please submit a bug report to https://github.com/modularml/mojo/issues and include the crash backtrace along with all the relevant source codes.
Stack dump:
0.      Program arguments: mojo u.mojo
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  mojo      0x000055dbd8d9f957
1  mojo      0x000055dbd8d9d52e
2  mojo      0x000055dbd8da002f
3  libc.so.6 0x00007f62b71b9520
Segmentation fault
```
My guess is that current Mojo version can't interoperate with a newer version of Python than what is based of, but older versions are compatible.

A comment from a Modular dev will be nice to clarify this issue <3

**Comment by johnhuichen** [comment]
Thanks @gujuriu. I tried your setup and that worked. What I did was:

1. Create a new environment using conda
```
conda create --name test-mojo-python-3.9 python=3.9
conda activate test-mojo-python-3.9
```
2. pip install tokenizers
3. link Python to Mojo
```
export MOJO_PYTHON_LIBRARY=~/anaconda3/envs/test-mojo-python-3.9/lib/libpython3.so
```

Since Anaconda installs python to a custom location, I wonder if it's because mojo was reading python libraries from ubuntu's default python. Your debug message says `No such file or directory` which also suggests something in that direction.
